src/core/map/tile.rs

```diff
@@ -166,3 +166,127 @@ impl<IO: EngineIO> TileTypes<IO> {
 	// 	Ok(idx as TileIdx)
 	// }
 }
+
+#[cfg(test)]
+mod tile_tests {
+	use super::*;
+	use proptest::prelude::*;
+	use std::{convert::Infallible, hash::Hasher, path::PathBuf};
+
+	#[derive(Debug, Default, Eq, PartialEq)]
+	struct DummyIO {}
+
+	impl EngineIO for DummyIO {
+		type ReadError = Infallible;
+		type Read = &'static [u8];
+
+		fn read(&mut self, _: PathBuf) -> Result<Self::Read, Self::ReadError> {
+			Ok(b"")
+		}
+
+		type TileInterface = ();
+
+		fn blank_tile_interface() -> Self::TileInterface {}
+
+		type TileAddedError = Infallible;
+
+		fn tile_added(
+			&mut self,
+			_: usize,
+			_: &mut TileType<Self>,
+		) -> Result<(), Self::TileAddedError> {
+			Ok(())
+		}
+	}
+
+	impl PartialEq for TileType<DummyIO> {
+		fn eq(&self, other: &Self) -> bool {
+			self.name == other.name
+		}
+	}
+	impl Eq for TileType<DummyIO> {}
+
+	impl std::hash::Hash for TileType<DummyIO> {
```

Comment:
Good point, I wonder if it's useful to enforce `Hash` on the `EngineIO` trait's implementations, but would it actually need to be hashed anywhere in the engine, hmm...  thoughts....